jta/src/main/java/io/smallrye/context/jta/context/propagation/JtaContextProvider.java

```diff
@@ -117,17 +118,23 @@ private TransactionManager tm() {
         if (lifecycleManagers.isResolvable()) {
             lifecycleManagers.get().setProvider(this);
         }
-        tm = CDI.current().select(TransactionManager.class).get();
-        if (tm != null) {
-            // validate it, because it can be an empty proxy
-            try {
-                tm.getStatus();
-            } catch (CreationException | SystemException x) {
-                x.printStackTrace();
-                // not really there
-                transactionManagerNotAvailable = true;
-                return null;
+        try {
+            tm = CDI.current().select(TransactionManager.class).get();
+            if (tm != null) {
+                // validate it, because it can be an empty proxy
```

Comment:
So, I don't know JTA internals, nor have I written this code. However, from CDI perspective, the only way you can ever get `null` is when a bean is created via a producer with `@Dependent` scope (see [this spec part](https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#producer_method)). Maybe Narayana does this?

In quarkus I don't think it can be the case because I found this class - https://github.com/quarkusio/quarkus/blob/main/extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/CDIDelegatingTransactionManager.java
Which I presume is the CDI-injected bean of TM.
Fun fact - in either of the cases i described above, the bean cannot be an "empty proxy" (at least not CDI proxy) because neither dependent not singleton beans use proxies :-)

Reply:
Hah, I am rather sure it used to be an empty proxy, otherwise, why would we have gotten this code?

Reply:
No idea, just saying that from CDI perspective this doesn't make much sense. But JTA could have been passing around an "empty object" that only worked under certain conditions. Asking someone from Narayana might shed more light.

Reply:
I don't dare take it out, it's too specific to have come up of my own imagination ;) I did add a comment though.